### PR TITLE
Read Publications from Relays

### PR DESCRIPTION
**Pull request by buttercat1791**
[ALEX-53](https://shadowysupercoding.atlassian.net/browse/ALEX-53?atlOrigin=eyJpIjoiNDE4ZDI0NjhkOGQ4NDNjN2JjZTljZjJjNmNmOWYyMGMiLCJwIjoiaiJ9)

This PR updates the Reader view to support AsciiDoc content as specified by NIP-62.

## Changes

- Add methods to the Pharos parser to fetch index and zettel events from relays and compose them into AsciiDoc documents.
- Reuse the Preview component to render publications.
- Trim Reader view support to just 30040 and 30041 event kinds.

## Testing

From the app's home page feed, click into one of the Aesop's Fables test publications, and confirm that it renders correctly.  

View the Gherkin scenarios in the Jira ticket for a full description of test cases.

## Notes for Reviewer

This branch is based on the branch for #2, so as to use the static site adapter.

[ALEX-53]: https://shadowysupercoding.atlassian.net/browse/ALEX-53?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by SilberWitch** [comment]
Tested with
- [Aesop's Fables](njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqstmt850zydulud34cqdhmavklr5l8de778j3h0rmjlv2dt3520j2cezz3aa): This is from my publishing tool. The original document contains = headers, but otherwise is plain-text. **ToC is disappeared (on all test events).**
- [The Tale of Peter Rabbit and Others (horiz.)](https://njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqsyr6lggn7yyjdarcquj387cfweum3qhl75mjxx82ymnv53r2ph90q0k6pvg): This is an audiobook and looks quite nice. **I don't know why the second cover pic doesn't show up, though, and there is no word-wrapping to the screen size, so the text runs off the right side of the page.**
- [The Tale of Peter Rabbit and Others](njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqs97hr6rwselerqzz8punqqzugv0dnh6ue8cj46q37hlj2d3ekp90s6rfs4x): This is the vertical view of the audiobook. **The text _audio file_ is a hyperlink, but that isn't identifiable from the rendering, which is confusing.**
- [Foods That Will Win The War And How To Cook Them: Chapter 1](njump.me/nevent1qvzqqqr4tqpzplfq3m5v3u5r0q9f255fdeyz8nyac6lagssx8zy4wugxjs8ajf7pqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqs8p09z542z847grl89xksd0slz9z8y874lucfc9jsqx5etwvvw8tccltr2j): Renders correctly.
- [Document Test](njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqs065tvd9g0h28zce8xynqgczst2s0am2q826gsvhxnst90qpzhs9czmx5lu): **Test of basic Asciidoc rendering doesn't show much rendering.**

**Review by Silberengel — CHANGES_REQUESTED** [review]
Tested with
- [Aesop's Fables](https://njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqstmt850zydulud34cqdhmavklr5l8de778j3h0rmjlv2dt3520j2cezz3aa): This is from my publishing tool. The original document contains = headers, but otherwise is plain-text. **ToC is disappeared (on all test events, might just be related to the PR2).**
- [The Tale of Peter Rabbit and Others (horiz.)](https://njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqsyr6lggn7yyjdarcquj387cfweum3qhl75mjxx82ymnv53r2ph90q0k6pvg): This is an audiobook and looks quite nice. **I don't know why the second cover pic doesn't show up, though, and there is no word-wrapping to the screen size, so the text runs off the right side of the page.**
- [The Tale of Peter Rabbit and Others](https://njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqs97hr6rwselerqzz8punqqzugv0dnh6ue8cj46q37hlj2d3ekp90s6rfs4x): This is the vertical view of the audiobook. **The text _audio file_ is a hyperlink, but that isn't identifiable from the rendering, which is confusing.**
- [Foods That Will Win The War And How To Cook Them: Chapter 1](https://njump.me/nevent1qvzqqqr4tqpzplfq3m5v3u5r0q9f255fdeyz8nyac6lagssx8zy4wugxjs8ajf7pqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqs8p09z542z847grl89xksd0slz9z8y874lucfc9jsqx5etwvvw8tccltr2j): Renders correctly.
- [Document Test](https://njump.me/nevent1qvzqqqr4tqpzq4ekxjmysc6vhtgs7fz3wasgn63ppyxegplhzh5rc4mmgcg6umkuqydhwumn8ghj7argv43kjarpv3jkctnwdaehgu339e3k7mgprpmhxue69uhhyetvv9ujumn0wdmksetjv5hxxmmdqqs065tvd9g0h28zce8xynqgczst2s0am2q826gsvhxnst90qpzhs9czmx5lu): **Test of basic Asciidoc rendering doesn't show much rendering.**

File: src/lib/components/Article.svelte [hunk 1]
```diff
@@ -104,62 +92,54 @@
...
+            {/each}
+          </SidebarGroup>
+        </SidebarWrapper>
+      </Sidebar>
+    {/if} -->
+    <div class='flex flex-col space-y-4 max-w-2xl'>
+      <Preview rootId={rootId} />
+    </div>
+  {:else}
+    <!-- TODO: Display empty state. -->
```
Comment (Silberengel) [inline src/lib/components/Article.svelte]: Ah, I see, now. Ignore my previous comment about the ToC.

Comment (buttercat1791) [reply src/lib/components/Article.svelte]: Yes we'll tackle ToC with [ALEX-50](https://shadowysupercoding.atlassian.net/browse/ALEX-50?atlOrigin=eyJpIjoiMDI3MmRmMTcyZDMyNGYyMjhiMzVjNWQ5OTMwZWU0YjkiLCJwIjoiaiJ9).

**Comment by buttercat1791** [comment]
It looks like I definitely need to add line wrapping in the final render, as well as make sure we're displaying images correctly.

I did see the audiobooks displaying nicely when I was testing it myself, but the links to the actual audio files aren't active.  That may be beyond the scope of this story, though.

**Comment by Silberengel** [comment]
The links actually worked. They just weren't styled like links, so I at first didn't think to click on them. They just look like plain text. Only some of the rendering is active, like for bold text and images. Everything else is just plain-text.

**Comment by Silberengel** [comment]
I'll make a second story, as a follow-on to this one, as I don't to hold up the merge with stylesheet schnick-schnack. I think adding the word-wrap is enough, for the first story.

I'll adjust the tickets.

**Review by Silberengel — APPROVED** [review]

**Review by Silberengel — CHANGES_REQUESTED** [review]
Decided word-wrap is formatting and moved it to the other ticket, but I think we need an auto-refresh or something here. I always have to refresh, to see the content.

**Comment by buttercat1791** [comment]
Taking a look at the refresh thing.

**Comment by buttercat1791** [comment]
I've fixed the word wrapping issue.

**Comment by buttercat1791** [comment]
@SilberWitch it appears that the app loads to a blank screen when it is unable to find some of the events needed to render the chosen publication.  I can add a "Not Found" page that prompts the user to refresh.

**Comment by Silberengel** [comment]
I'm not seeing a notification. Just stays blank.

**Comment by buttercat1791** [comment]
Yep, same for me.  That commit includes the message, but the message doesn't show.  More work to be done.

**Comment by Silberengel** [comment]
- I now see no pictures at all, in the horizontal audiobook.
- The word-wrap seems to work everywhere, except in verbatim quotes (which is correct).
- When quotes run off the page, like with the big drawings the Bitcoin Whitepaper has, and I scroll to the right, then the background color drops off.
image
2) The tables shouldn't word-wrap (correct), but the text inside of tables shouldn't run outside of the bounds of the table and the width of the table should be relative to the screen. You can see tables in the [horizontal audiobook](https://njump.me/nevent1qqszkl2d2pcu5du4fussfq05qxsc6fdvz5afcvz72n02yhzs5rvdjsszyptnvd9kfp35ewk3puj9zamq384zzzgdjsrlw90g83thk3s34ehdcauw9mm).
These are tables:
image

**Comment by Silberengel** [comment]
The message would show for every index event, apparently. :shrug: Makes it seem permanently broken.

**Comment by buttercat1791** [comment]
I think the issue with verbatim quotes and diagrams running off the screen should be resolved in the follow-up work for improving the stylesheets.  What I'd like to do is have diagram blocks be side-scrollable, but the visible area stays within the bounds of the main content column.

**Comment by Silberengel** [comment]
I'll add that to the other issue ticket, then.

**Comment by Silberengel** [comment]
Then it's just the issue with the blank screens left.

**Comment by buttercat1791** [comment]
Yup.  I'm seeing `500` errors when I try to load articles, too.  Gonna be investigating that, but we can't merge until it is fixed.

**Review by Silberengel — APPROVED** [review]

**Comment by Silberengel** [comment]
Bam! So smooth. Merging it.

**Pull request merged**